src/DocumentDB.ChangeFeedProcessor/LeaseManagement/DocumentServiceLease.cs

```diff
@@ -32,9 +35,17 @@ public DocumentServiceLease(DocumentServiceLease other)
             this.Properties = other.Properties;
         }
 
-        [JsonProperty("id")]
+        [JsonProperty(IdPropertyName)]
         public string Id { get; set; }
 
+        /// <summary>
+        /// Gets or sets property to be used as partition key path for lease collections.
+        /// This is clone of existing Id property to maintain backward compat.
+        /// This property name is compatible to both GremlinAccounts and SqlAccounts
+        /// </summary>
+        [JsonProperty(LeasePartitionKeyPropertyName)]
```

Comment:
Let's make sure that when it's null, we don't serialize to JSON. I think that would be NullValueHandling property.